torch/_inductor/codegen/simd.py

```diff
@@ -1748,9 +1748,12 @@ def _bench(candidate_split_size):
                 partial_accum.reduction_type, partial_accum.reduction_type
             )
 
-            V.graph.wrapper_code.writeline(
-                f"{buffer_name} = {ws_name}[{start} : {end}].view({nsplit}, {rnumel}).{opname}(dim=0)",
-            )
+            final_reduce = f"{buffer_name} = {ws_name}[{start} : {end}].view({nsplit}, {rnumel}).{opname}(dim=0)"
```

Comment:
can we update the final_reduce so that we allocate it in the right number of bytes to begin with ? instead of a clone below

Reply:
Can we allocate temporary in fp32, write to it in fp32, then sum -> output in lower precision ?

Reply:
I'm not sure if eager would truncate the input first if we do `torch.sum(x, dtype=torch.bfloat16)` . On the other hand, these tensors are all very small since they are partially reduced tensors.